mzLib/Omics/BioPolymer/IHasSequenceVariants.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Omics.Modifications;
+namespace Omics.BioPolymer;
+
+/// <summary>
+/// Represents a BioPolymer capable of having many sequence variants. 
+/// Sequence variant parsing and construction is handled in <see cref="VariantApplication"/>
+/// </summary>
+public interface IHasSequenceVariants
+{
+    /// <summary>
+    /// Primary Sequence
+    /// </summary>
+    string BaseSequence { get; }
+
+    /// <summary>
+    /// Sample name from which applied variants came, e.g. tumor or normal.
+    /// </summary>
+    public string SampleNameForVariants { get; }
+
+    /// <summary>
+    /// Modifications (values) located at one-based protein positions (keys)
+    /// </summary>
+    IDictionary<int, List<Modification>> OneBasedPossibleLocalizedModifications { get; }
+
+    /// <summary>
+    /// Original modifications as defined in the Parsed XML database
+    /// </summary>
+    IDictionary<int, List<Modification>> OriginalNonVariantModifications { get; set; }
+
+    /// <summary>
+    /// Original BioPolymer used to construct all sequence variants. 
+    /// </summary>
+    IBioPolymer NonVariant { get; }
```

Comment:
how about consensus sequence. not sure that anything is nonvariant. just most common. uniprot and refseq with have different consensus sequences and variants for the same protiein

Reply:
changed to ConsensusVariant